lib/block_elements.ex

```diff
@@ -167,6 +173,13 @@ defmodule BlockBox.BlockElements do
   end
 
   def select_menu(placeholder, type, action_id, opts) do
+    init_option = Keyword.get(opts, :initial_option, false)
+    options = Keyword.get(opts, :options, false)
+    opts = 
+      case init_option != false and is_integer(init_option) and options != false do 
+        true -> Keyword.put(opts, :initial_option, Enum.at(options, init_option))  
+        false -> opts
+      end
```

Comment:
this is a poor use of a case statement. 
a cond statement or even an if statement would be much better here

Reply:
what warrants much better? Performance wise they are the same. If anything case is more consistent for usage. It's a matter of syntax if anything.

Reply:
abc = true and def = false in these benchmarks
<img width="1209" alt="Screen Shot 2019-12-05 at 5 27 27 PM" src="https://user-images.githubusercontent.com/20006218/70279491-95871000-1784-11ea-90fd-1c2fb9857c80.png">
![Uploading Screen Shot 2019-12-05 at 5.27.12 PM.png…]()
![Uploading Screen Shot 2019-12-05 at 5.26.59 PM.png…]()

Reply:
you're right, the syntax isn't bad, the code itself is bad. this is an example of stuffing an imperative style into a functional language. a real improvement would come from re-organizing the code so that you never need to do something like this in the first place.

Reply:
It's quite the opposite, if and cond is more imperative and cases are more functional because of their pattern-matching capabilities. Also speed-wise it's better to go the case way.

Reply:
you're missing the point. the way you set the variables "init_option" and "options" basically means you corner yourself into writing an ugly 3 term conditional statement. Just because you're using a case statement to evaluate a conditional doesn't mean your code is "more functional". All 3 options at this point are equally functional because all of them evaluate an ugly conditional.

In a truly functional solution, you wouldnt have gotten into this mess in the first place. This is what I meant. My pseudocode in the review is a much better example of functional style that avoids conditionals like the one you're doing.

Reply:
Case is unequivocally more functional when talking from a programming paradigm case. We are discussing the use of case over cond and if, not your general comment on the flow.